### PR TITLE
feat: make social buttons work

### PR DESCRIPTION
**Pull request by danielgent**
This works *BUT* how to test it?

I was thinking of loading the URL on each social button and then parsing the query string params and passing them to `/authorize` to check we log in :thinking:

### CODE REVIEW

File: src/utils/components/SocialButton.tsx [hunk 1]
```diff
@@ -1,40 +1,73 @@
...
+    queryString.set("response_type", session.authParams.response_type);
+  }
+  if (session.authParams.state) {
+    queryString.set("state", session.authParams.state);
+  }
+  const href = `/authorize?${queryString.toString()}`;
+
   return (
-    <Button
+    <a
```
Comment (danielgent) [inline src/utils/components/SocialButton.tsx:46]: In login2 this is React so I use a button which then calls `auth0.js` to build this URL *BUT* there's no need for that here

File: src/utils/components/SocialButton.tsx [hunk 2]
```diff
@@ -1,40 +1,73 @@
...
       className={cn(
+        "block",
+        // copied from Button.tsx as this component is now an anchor
+        "relative w-full rounded-lg text-center px-4 py-5 bg-primary text-textOnPrimary hover:bg-primaryHover",
```
Comment (danielgent) [inline src/utils/components/SocialButton.tsx:50]: an alternative would be to have the `<Button/>` component accept a `Component` prop where we can do `Component=a` but :shrug: 

For now I'd rather keep it simple :smile:

File: src/utils/components/SocialButton.tsx [hunk 3]
```diff
@@ -1,40 +1,73 @@
...
-      {icon || ""}
-      <div
-        className={cn("text-left text-black dark:text-white sm:text-base", {
-          ["hidden sm:inline short:hidden"]: canResize,
-        })}
-      >
-        {text}
-      </div>
-    </Button>
+      <span className="flex items-center justify-center space-x-2">
```
Comment (danielgent) [inline src/utils/components/SocialButton.tsx:60]: seems like we can use `className` *OR* `class` and both work :shrug:

File: test/integration/helpers/playwrightSnapshots.ts [hunk 4]
```diff
@@ -7,7 +7,25 @@ import { EmailOptions } from "../../../src/services/email/EmailOptions";
...
+    width: 1920,
+    height: 1080,
+  },
+  md: {
+    width: 1280,
+    height: 720,
+  },
+  sm: {
+    width: 375,
+    height: 667,
```
Comment (danielgent) [inline test/integration/helpers/playwrightSnapshots.ts:21]: we aren't catching responsiveness with the default browser size

**Review by markusahlstrand — APPROVED** [review]
Looks good. I'm thinking that this should be dynamic and that we switch to use the arctic project to do the actual auth. Think validating the url's are just fine until we have playwright tests running on it.

File: src/utils/components/SocialButton.tsx (hunk 1 above)
Comment (markusahlstrand) [reply src/utils/components/SocialButton.tsx:46]: I guess an a-tag is correct from a a11y-perspecitve?

Comment (danielgent) [reply src/utils/components/SocialButton.tsx:46]: Not even a perspective. There's no clientside JS, so an anchor is the only way. A button is for submitting forms :sunglasses:

**Comment by danielgent** [comment]
> Looks good. I'm thinking that this should be dynamic and that we switch to use the arctic project to do the actual auth. Think validating the url's are just fine until we have playwright tests running on it.

Arctic project?  Whatever you think though Markus I'm just using the tried-and-tested social auth flow that we already have working :+1:

File: src/utils/components/SocialButton.tsx (hunk 1 above)
Comment (markusahlstrand) [reply src/utils/components/SocialButton.tsx:46]: Thinking if you have a screen reader for blind people? Think there's a difference between a form submit and a link right?

Comment (danielgent) [reply src/utils/components/SocialButton.tsx:46]: Check out NVDA or Jaws! :smile:

**Pull request merged**